homeassistant/components/generic/config_flow.py

```diff
@@ -410,3 +452,30 @@ async def async_step_init(
             ),
             errors=errors,
         )
+
+
+class CameraImagePreview(HomeAssistantView):
+    """Camera view to temporarily serve an image."""
+
+    url = "/api/generic/preview_flow_image/{flow_id}"
+    name = "api:generic:preview_flow_image"
+    requires_auth = False
```

Comment:
I was thinking more about how the user input for the preview is left around, and still a little uncomfortable with this and having a change of heart from before trying to do something simple/scrappy to instead just trying to really make the preview just go away at a reasonable time frame.  Set a timer and clear the preview? Then it removes the caveats just added and is the most straight forward to reason about, but requires a little more code.

Apologies for all the back and forth on this, it's just taken me awhile to wrap my head around everything the camera view provides and I had forgotten that this disables auth checks.

Reply:
Ok no problem.

How about an expiry time e.g. +5mins. Then if a GET comes in after that, we ignore it, possibly even delete it at that point.

That way it doesn't need a timer process but limits the exposure.

Reply:
Yeah, true, a timer would also keep the config flow alive after its finished.

You could also clear it in the final step explicitly too.

Reply:
@allenporter Done now.